OpenDreamShared/Compiler/DM/DMParser.cs

```diff
@@ -12,197 +12,156 @@ public partial class DMParser : Parser<Token> {
-                            }
-                        } else {
-                            //Var override
+                            DMASTExpression value;
                             if (Check(TokenType.DM_Equals)) {
                                 Whitespace();
-                                DMASTExpression value = Expression();
+                                value = Expression();
                                 if (value == null) Error("Expected an expression");
+                            } else {
+                                value = new DMASTConstantNull();
+                            }
+
+                            AsTypes();
 
-                                statement = new DMASTObjectVarOverride(_currentPath, value);
+                            varDefinitions.Add(new DMASTObjectVarDefinition(varPath, value));
+                            if (Check(TokenType.DM_Comma)) {
+                                Whitespace();
+                                DMASTPath newVarPath = Path();
+                                if (newVarPath == null) Error("Expected a var definition");
+                                if (newVarPath.Path.Elements.Length > 1) Error("Invalid var name"); //TODO: This is valid DM
+
+                                varPath = _currentPath.AddToPath("../" + newVarPath.Path.PathString);
                             } else {
-                                //Empty object definition
-                                statement = new DMASTObjectDefinition(_currentPath, null);
+                                break;
                             }
                         }
+
+                        if (varDefinitions.Count == 1) {
+                            statement = varDefinitions[0];
+                        } else {
+                            statement = new DMASTMultipleObjectVarDefinitions(varDefinitions.ToArray());
+                        }
                     }
-                }
 
-                _currentPath = oldPath;
-                return statement;
+                    //Var override
+
+                    if (statement == null && Check(TokenType.DM_Equals)) {
+                        Whitespace();
+                        DMASTExpression value = Expression();
+                        if (value == null) Error("Expected an expression");
+
+                        statement = new DMASTObjectVarOverride(_currentPath, value);
+                    }
+
+                    //Empty object definition
+                    if (statement == null) {
+                        statement = new DMASTObjectDefinition(_currentPath, null);
+                    }
+
+                    if (!PeekDelimiter() && Current().Type != TokenType.DM_Dedent) {
+                        Error("Expected end of statement");
+                    }
+
+                    _currentPath = oldPath;
+                    return statement;
+                } catch (CompileErrorException) {
+                    _currentPath = oldPath;
```

Comment:
```suggestion

```

Reply:
If you didn't know, you can make a multiple-line suggestion by click-dragging on the lines when selecting them.